### DISCUSSION TITLE
QROptions expected type mismatch flagged

### DISCUSSION THREAD

**Discussion by danieldevine** [chillerlan/php-qrcode#112] [Q&A]
Hi,

This doesn't break anything, all works as expected, but thought you might like to know. 

When using QROptions, following an example in the docs, the following issue is flagged by intellephense when QROptions are passed to a new QRCode. 

 "Expected type 'chillerlan\\Settings\\SettingsContainerInterface|null'. Found 'chillerlan\\QRCode\\QROptions'.",

```php
        $options = new QROptions([
            'version'    => 5,
            'outputType' => QRCode::OUTPUT_MARKUP_SVG,
            'eccLevel'   => QRCode::ECC_L,
        ]);

        $qr =  (new QRCode($options))->render($data);
```

 - PHP version/OS: 7.4.12/MacOS 12.1
 - Library version: 4.3.2

**Comment by codemasher** [comment]
Hmm, looks like Intelephense doesn't properly analyze the ancestors of the `QROptions` class - that is, in fact, a [`SettingsContainerInterface`](https://github.com/chillerlan/php-qrcode/blob/6356b246948ac1025882b3f55e7c68ebd4515ae3/src/QRCode.php#L153-L158) - is it perhaps a configuration issue? For phan it's possible to [configure the `vendor` path](https://github.com/chillerlan/php-qrcode/blob/6356b246948ac1025882b3f55e7c68ebd4515ae3/.phan/config.php#L19-L32), so that it will be registered during runtime.
